app/src/main/java/com/example/zingmp3phake/screen/detailplaylist/DetailPlaylistActivity.kt

```diff
@@ -1,17 +1,131 @@
 package com.example.zingmp3phake.screen.detailplaylist
 
+import android.app.PendingIntent
+import android.content.ContentUris
+import android.content.Intent
+import android.content.IntentFilter
+import android.net.Uri
+import android.view.View
+import androidx.core.net.toUri
 import com.example.zingmp3phake.R
+import com.example.zingmp3phake.data.model.Song
 import com.example.zingmp3phake.databinding.ActivityDetailPlaylistBinding
+import com.example.zingmp3phake.screen.detailsong.DetailSongActivity
+import com.example.zingmp3phake.screen.personal.RecyclerViewRecentAdapter
+import com.example.zingmp3phake.utils.ACTION_MUSIC
+import com.example.zingmp3phake.utils.ACTION_MUSIC_BROADCAST
+import com.example.zingmp3phake.utils.BUNDLE_LIST_KEY
+import com.example.zingmp3phake.utils.BUNDLE_TITLE_STRING_KEY
+import com.example.zingmp3phake.utils.DATA_KEY
+import com.example.zingmp3phake.utils.MEDIA_EXTERNAL_AUDIO_URI
+import com.example.zingmp3phake.utils.MusicAction
+import com.example.zingmp3phake.utils.SONG
 import com.example.zingmp3phake.utils.base.BaseActivity
+import com.example.zingmp3phake.utils.loadByGlide
+import kotlin.random.Random
 
 class DetailPlaylistActivity :
-    BaseActivity<ActivityDetailPlaylistBinding>(ActivityDetailPlaylistBinding::inflate) {
+    BaseActivity<ActivityDetailPlaylistBinding>(ActivityDetailPlaylistBinding::inflate),
+    RecyclerViewRecentAdapter.ItemClickListener,
+    DetailPlaylistContract.View {
 
+    private val adapterRV = RecyclerViewRecentAdapter(this)
+    private val detailPlaylistPresenter = DetailPlaylistPresenter(this)
     override fun initData() {
-        // TODO later
+        val mIntent = intent
+        val bundle = mIntent.getBundleExtra(DATA_KEY)
+        val title = bundle?.getString(BUNDLE_TITLE_STRING_KEY)
+        val list = bundle?.getParcelableArrayList<Song>(BUNDLE_LIST_KEY)
+        binding.textviewNumberOfListSong.text = "${list?.size} $SONG"
+        binding.recyclerView.adapter = adapterRV
+        if (list != null) adapterRV.setData(list)
+        if (title != null) binding.textviewPlaylistName.text = title
+
+        detailPlaylistPresenter.apply {
+            bindService(applicationContext)
+            val filter = IntentFilter(ACTION_MUSIC_BROADCAST)
+            registerReceiver(localReciver, filter)
+        }
```

Comment:
```suggestion
        detailPlaylistPresenter.apply {
            bindService(applicationContext)
            val filter = IntentFilter(ACTION_MUSIC_BROADCAST)
            registerReceiver(localReciver, filter)
        }
```
Move thằng này vô hàm ``handleEvents`` đi e,
Nó không thuộc về ``initData()``